pkgs/by-name/da/dashy-ui/package.nix

```diff
@@ -0,0 +1,57 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  fetchYarnDeps,
+  yarnConfigHook,
+  yarnBuildHook,
+  nodejs,
+  yq-go,
+  settings ? { },
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "dashy-ui";
+  # This is like 3.1.1 but the latest working yarn.lock.
+  # All other changes are for docs with the exception of 768d746cbfcf365c58ad1194c5ccc74c14f3ed3a, which simply adds no-referrer meta tag
+  version = "0b1af9db483f80323e782e7834da2a337393e111";
+  src = fetchFromGitHub {
+    owner = "lissy93";
+    repo = "dashy";
+    rev = "${finalAttrs.version}";
+    hash = "sha256-lRJ3lI9UUIaw9GWPEy81Dbf4cu6rClA4VjdWejVQN+g=";
+  };
+  yarnOfflineCache = fetchYarnDeps {
+    yarnLock = finalAttrs.src + "/yarn.lock";
+    hash = "sha256-KVAZIBM47yp1NWYc2esvTwfoAev4q7Wgi0c73PUZRNw=";
+  };
+  # - If no settings are passed, use the default config provided by upstream
+  # - Despite JSON being valid YAML (and the JSON passing the config validator),
+  # there seem to be some issues with JSON in the final build - potentially due to
+  # the way the client parses things
+  # - Instead, we use `yq-go` to convert it to yaml
+  # Config validation needs to happen after yarnConfigHook, since it's what sets the yarn offline cache
+  postYarnConfigHook = lib.optional (settings != { }) ''
```

Comment:
One benefit of doing the config check is that when updating to a new package version it should fail to build before doing a full compile, which can take a bit even on powerful/recent hardware

Reply:
I think this is up to your discretion, since running the config check regardless would basically be like implementing a checkPhase test. It could be nice to spot broken releases/commits.

Reply:
I think I'll leave it for now and do some more hacking later. I'll probably look into the ordering of phases (because of the yarn cache) when there's a new full release upstream, and maybe change it then.